pkg/security/security.go

```diff
@@ -124,27 +119,15 @@ type Options struct {
 	// secret TTL.
 	SecretTTL time.Duration
 
-	// The initial backoff time in millisecond to avoid the thundering herd problem.
-	InitialBackoffInMilliSec int64
-
-	// secret should be rotated if:
-	// time.Now.After(<secret ExpireTime> - <secret TTL> * SecretRotationGracePeriodRatio)
+	// The ratio of cert lifetime to refresh a cert. For example, at 0.10 and 1 hour TTL,
+	// we would refresh 6 minutes before expiration.
 	SecretRotationGracePeriodRatio float64
 
-	// Key rotation job running interval.
-	RotationInterval time.Duration
```

Comment:
IIRC, this RotationInterval was 5 minutes, so that in the grace period, if the cert refreshing fails, it will re-refresh 5 minutes later. Do we have a similar logic that could handle the cert refresh scenario?

Reply:
Interesting, I hadn't considered the refresh failing. Let me see if I can add a test for this case. However, what I expect is:

* Cert refresh is triggered, we send an update to sds server.
* SDS server calls GenerateSecret, which fails. The XDS server returns error up to envoy, causing envoy to disconnect.
* Envoy will repeatedly reconnect to sds server and re-attempt GenerateSecret until the CA is back online

So the end result is we are resilient to CA downtime during refresh

Reply:
I added a test case for this

Reply:
By reading the changes in server.go and secretcache.go, the rotation task will clear the cache for `default`, and then call Server.UpdateCallback(`default`), which calls `Push` and adds a `PushRequest`. 
sdsservice.Generate()->sdsservice.generate()->SecretManagerClient.GenerateSecret(). If refresh fails, then pushes error to envoy and close the stream.

Does this new flow support multiple concurrent SDS streams? Do we have tests covering that?

Reply:
Yes, we do support this. That is one of the huge benefits of the change, we are use the battle-tested Istiod XDS server which has dealt with every obscure XDS edge case over the past few years. Also we have an explicit test for this in `TestSDS`

Reply:
Are there any README or docs describing the Istiod XDS server? That would help us to better understand and maintain the adoption of XDS server in other places outside Istiod.

Reply:
Unfortunately no. I have been meaning to make one similar to the README in this PR, but haven't yet